debezium-core/src/main/java/io/debezium/relational/RelationalChangeRecordEmitter.java

```diff
@@ -171,16 +171,15 @@ protected boolean skipEmptyMessages() {
     protected void emitUpdateAsPrimaryKeyChangeRecord(Receiver<P> receiver, TableSchema tableSchema, Struct oldKey,
                                                       Struct newKey, Struct oldValue, Struct newValue)
             throws InterruptedException {
+        Struct deleteEnvelope = tableSchema.getEnvelopeSchema().delete(oldValue, getOffset().getSourceInfo(), getClock().currentTimeAsInstant());
```

Comment:
@vjuranek How about being really sure and call `getOffset().getSourceInfo()` only once into a variable and re-use that variable? Also in the subsequent call there is `getOffset` called twice. Maybe it should be called only once at the top of method too  just in case?

Reply:
Right, originally I thought `getOffset().getSourceInfo()` returns a reference to `SourceInfo` object which is mutable, but you are right, it actually returns `Struct` with current values.